### PR TITLE
Issue6

### LINKED ISSUE #6: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by tmcswain**
Closes #6

### CODE REVIEW

**Comment by tmcswain** [comment]
[Issue 6: Image Tags](https://trello.com/c/nwIUgb4K/278-issue-6-image-tags)

File: .rubocop.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ AllCops:
     - 'vendor/**/*'
     - 'node_modules/**/*'
     - 'bin/*'
+    - 'db/migrate/*.acts_as_taggable_on_engine.rb'
```
Comment (natalieschauser) [inline .rubocop.yml]: Can we add this to the previous commit? It seems more pertinent to that one :)

File: app/views/images/show.html.erb [hunk 2]
```diff
@@ -1 +1,6 @@
-<img src="<%= @image.url %>">
+<%= image_tag @image.url %>
```
Comment (natalieschauser) [inline app/views/images/show.html.erb]: I think the `show` changes can be pulled out into it's own commit to make the commits more atomic (https://www.freshconsulting.com/atomic-commits/)

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -26,14 +26,29 @@ def test_create_with_valid_url_displays_img
     assert_equal img_url, image.url
   end
 
-  def test_show
+  def test_create__adds_tags
+    img_url = 'https://embedwistia-a.akamaihd.net/deliveries/6aed31ef61cf95f5caad5d1a028bc7a06ce0f994.jpg?image_crop_resized=1280x720'
+    assert_difference %w[Image.count Image.tagged_with("flower").count Image.tagged_with("pink").count], 1 do
+      post images_url,
+           params: { image: { url: img_url, tag_list: 'flower, pink' } }
```
Comment (natalieschauser) [inline test/controllers/images_controller_test.rb]: What happens if the user enters in `flower, pink,,,,,`. Does `acts-as-taggbe` handle that or do we need to handle that?

Comment (natalieschauser) [inline test/controllers/images_controller_test.rb]: What if the user doesn't enter in any tags. What do the params look like? We might need to update the existing `test_create_...` tests to reflect this new parameter.
